### PR TITLE
ENGDX 459: upgrade to node 20 and bumped dependencies 

### PR DESCRIPTION
**Pull request by AdamTranquilla**
- upgrade to node 20
- bumped dependencies
- updated lint

``` .md
{
  "dependencies": {
    "@actions/core": "^1.6.0 -> ^1.10.1",
    "@actions/github": "^5.0.0 -> ^6.0.0",
    "@octokit/action": "^3.18.0 -> ^6.0.7",
    "axios": "^0.25.0 -> ^1.6.7",
    "is-plain-object": "^5.0.0",
    "lcov-parse": "^1.0.0"
  },
  "devDependencies": {
    "@types/jest": "^27.4.0 -> ^29.5.12",
    "@types/lcov-parse": "^1.0.0 -> ^1.0.2",
    "eslint": "^7.17.0 -> ^8.56.0",
    "eslint-config-neo": "^0.6.3 -> ^0.9.0",
    "jest": "^27.5.0 -> ^29.7.0",
    "prettier": "^2.5.1 -> ^2.8.8",
    "rimraf": "^3.0.2 -> ^5.0.5",
    "ts-jest": "^27.1.3 -> ^29.1.2",
    "typescript": "^4.5.5 -> ^5.3.3"
  }
}
```

### CODE REVIEW

File: src/get-data.ts [hunk 1]
```diff
@@ -39,10 +39,10 @@ const getData = async (authToken?: string): Promise<PullRequestData> => {
     prData.actor = info.sender.login;
   } else if (!info.repository && !info.sender) {
     throw new Error('repository and sender are undefined');
-  } else if (!info.repository) {
-    throw new Error('repository is undefined');
-  } else {
+  } else if (info.repository) {
```
Comment (leah-is-offline) [inline src/get-data.ts:42]: why

File: src/make-comment.ts [hunk 2]
```diff
@@ -39,18 +39,18 @@ const makePullRequestComment = async (message: string | undefined, commentData:
...
     const body = message ?? (await constructComment(commentData));
 
-    if (!botComment) {
-      octokit.issues.createComment({
+    if (botComment) {
+      octokit.issues.updateComment({
         owner: owner,
         repo: repo,
-        issue_number: pullRequestNumber,
+        comment_id: botComment.id,
```
Comment (leah-is-offline) [inline src/make-comment.ts:46]: also why

File: src/get-data.ts (hunk 1 above)
Comment (AdamTranquilla) [reply src/get-data.ts:42]: eslint bump complained until I swapped it

File: src/make-comment.ts (hunk 2 above)
Comment (AdamTranquilla) [reply src/make-comment.ts:46]: same as above! the order of the if/else is swapped not the fields inside each block

File: package.json [hunk 3]
```diff
@@ -25,24 +25,24 @@
...
   "devDependencies": {
-    "@types/jest": "^27.4.0",
-    "eslint": "^7.17.0",
-    "eslint-config-neo": "^0.6.3",
-    "jest": "^27.5.0",
-    "prettier": "^2.5.1",
-    "rimraf": "^3.0.2",
-    "ts-jest": "^27.1.3",
-    "typescript": "^4.5.5"
+    "@types/jest": "^29.5.12",
```
Comment (leah-is-offline) [inline package.json:39]: check if these support running -- --coverage the same 

the action relies on the --coverage file being generated and the build looks like it cant find it. which makes me think its not being generated. 

if you pull this locally and run `npm run test -- --coverage` and an lcov.info file is generated than this is not the problem
if it is not generated, then this is likely the problem

Comment (AdamTranquilla) [reply package.json:39]: It passed locally :(

Comment (leah-is-offline) [reply package.json:39]: using node 20?

Comment (AdamTranquilla) [reply package.json:39]: Yup! everything works locally

Comment (AdamTranquilla) [reply package.json:39]: on this version of this branch

**Review by iansu — APPROVED** [review]

**Review by neo-bryce-brandford — APPROVED** [review]

**Pull request merged**